spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/TaskRepository.java

```diff
@@ -27,23 +30,26 @@
 public interface TaskRepository {
 
 	/**
-	 * Notifies the repository that a taskExecution needs to be updated.
+	 * Notifies the repository that a taskExecution has completed.
 	 *
-	 * @param taskExecution taskExecution to be updated
+	 * @param executionId to the task execution to be updated.
+	 * @param exitCode to be stored for this task.
+	 * @param endTime designated when the task completed.
+	 * @param exitMessage to be stored for the task.
 	 */
-	public void update(TaskExecution taskExecution);
+	@Transactional
+	public void completeTaskExecution(long executionId, Integer exitCode, Date endTime,
+			 String exitMessage);
```

Comment:
With this, we update the database, but don't update the underlying object (the `TaskExecution`) which leaves that update to the consumer of this API.  Thoughts on returning the updated `TaskExecution`?